src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -385,6 +385,11 @@ public int CompareTo(object? obj)
         /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
         public int CompareTo(Half other)
         {
+            if (this == other)
```

Comment:
It's not clear why this fix is correct.

`float` and `double` are both implemented as `<, >, ==, NaN` (how the code is prior to this PR), which leads me to believe that `<` or `>` are incorrectly handling an equals case.

Reply:
@tannergooding What do you think we should do on this? I saw your comment above, but I wasn't sure if you were hoping @ntovas could share more info or if you had other ideas.

Reply:
As is, I don't believe this is the correct fix. It needs someone (either the PR author, myself, or @pgovind) to do the further investigation as to where the actual bug is and where the `CompareTo` logic (which is identical to what `float`/`double` do) is failing.

Reply:
@ntovas Do you think you'd be able to investigate this further?